### PR TITLE
Backend project structure

### PR DESCRIPTION
**Pull request by ranta**

### CODE REVIEW

**Review by toikarin-vincit — APPROVED** [review]
as we're on the very early stages of development, no need to consider these comments as blocking

File: backend/README.md [hunk 1]
```diff
@@ -1 +1,48 @@
-Hitas backend for City of Helsinki
+# Hitas backend for City of Helsinki 
+
+## Prerequisites
+
+* [Python 3.8](https://www.python.org/)
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/README.md:5]: I guess 3.10 _should_ work as well, I was a bit worried about some dependencies not supporting it yet 🤔

[the remark this answers is not in the document]
Comment (toikarin-vincit) [reply backend/README.md:5]: The non-backwards compatible changes in 3.9 and 3.10 seems a like a quite short list to me. I tested this branch with `3.10.4` (which is default python3 version in Ubuntu 22.04 LTS)

[the remark this answers is not in the document]
Comment (ranta) [reply backend/README.md:5]: Oki doki, lets switch to `3.10` :)

File: backend/docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# Collect static files
+echo "Collecting static files..."
+./manage.py collectstatic --noinput
+
+# Translate messages
+echo "Updating translations..."
+./manage.py compilemessages -l fi
+
+echo "Checking for odd ENTRYPOINT line in arguments"
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/docker-entrypoint.sh:30]: I kept it case we run into same problems in Azure that it was trying to solve here:
https://github.com/City-of-Helsinki/city-infrastructure-platform/commit/4f2480cab0e8eb919324cdfbed6ffd82f90a0061

I will remove it for now, lets see if there's any problems later :)

File: backend/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+version: "3.8"
+services:
+  hitas-db:
+    image: postgres:12
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/docker-compose.yml:4]: Bumped Postgres to 13 everywhere, as that's the version mentioned by Platta people

File: backend/format.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+cmd='autoflake --recursive --in-place --remove-all-unused-imports --ignore-init-module-imports . && isort . && black . && flake8 .'
+
+if [ -f /.dockerenv ]; then
+    eval "$cmd"
+else
+    docker-compose run --no-deps --rm --entrypoint='/bin/sh -c' hitas "$cmd"
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/format.sh:8]: Idea here was that the required packages might not exist locally (or in virtualenv) if user is using docker.

Want me to change this to be executed locally?

File: backend/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /var/cache/apt/archives && \
+    rm -rf /root/.cache/pypoetry && \
+    rm -rf /root/.cache/pip
+
+COPY docker-entrypoint.sh /usr/local/bin
+ENTRYPOINT ["docker-entrypoint.sh"]
+
+# ==============================
+FROM base AS development
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/Dockerfile:39]: Maybe 🤔 Let's improve the Dockerfile in the future if necessary

File: backend/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ENV DEV_SERVER=1
+
+RUN /root/.local/bin/poetry install
+COPY . /hitas-backend
+RUN chown -R appuser:appuser /hitas-backend
+USER appuser
+EXPOSE 8000
+
+# ==============================
+FROM base AS production
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/Dockerfile:54]: Thats could be another way of doing it, I don't really have a preference for either

File: backend/docker-entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -e
+
+echo "Version: $VERSION"
```
[the remark this answers is not in the document]
Comment (ranta) [reply backend/docker-entrypoint.sh:5]: Resolved any errors it raised ✅

**Review by toikarin-vincit — APPROVED** [review]

**Pull request merged**